src/node/internal/internal_tls_wrap.ts

```diff
@@ -0,0 +1,644 @@
+    validateBuffer(context, 'context');
+  }
+
+  throw new Error('exportKeyingMaterial is not implemented');
+};
+
+TLSSocket.prototype.setMaxSendFragment = function setMaxSendFragment(
+  this: TLSSocket,
+  size: number
+): boolean {
+  validateInt32(size, 'size');
+  // Setting maximum TLS fragment size is not supported.
+  return false;
+};
+
+TLSSocket.prototype._handleTimeout = function _handleTimeout(
+  this: TLSSocket
+): void {
+  this._emitTLSError(new ERR_TLS_HANDSHAKE_TIMEOUT());
+};
+
+TLSSocket.prototype._emitTLSError = function _emitTLSError(
+  this: TLSSocket,
+  err: Error
+): void {
+  const e = this._tlsError(err);
+  if (e) this.emit('error', e);
+};
+
+TLSSocket.prototype._tlsError = function _tlsError(
+  this: TLSSocket,
+  err: Error
+): Error | null {
+  this.emit('_tlsError', err);
+  if (this._controlReleased) return err;
+  return null;
+};
+
+TLSSocket.prototype._releaseControl = function _releaseControl(
+  this: TLSSocket
+): boolean {
+  if (this._controlReleased) return false;
+  this._controlReleased = true;
+  this.removeListener('error', this._tlsError.bind(this));
+  return true;
+};
+
+// This function is called from net.Socket onConnectionOpened() handler.
+TLSSocket.prototype._finishInit = function _finishInit(this: TLSSocket): void {
+  // Guard against getting onhandshakedone() after .destroy().
+  // * 1.2: If destroy() during onocspresponse(), then write of next handshake
+  // record fails, the handshake done info callbacks does not occur, and the
+  // socket closes.
+  // * 1.3: The OCSP response comes in the same record that finishes handshake,
+  // so even after .destroy(), the handshake done info callback occurs
+  // immediately after onocspresponse(). Ignore it.
+  if (!this._handle) return;
+
+  this._secureEstablished = true;
+  if (
+    this._tlsOptions.handshakeTimeout &&
+    this._tlsOptions.handshakeTimeout > 0
+  )
+    this.setTimeout(0, this._handleTimeout.bind(this));
+  this.emit('secure');
+};
+
+TLSSocket.prototype._start = function _start(this: TLSSocket): void {
+  if (this.connecting) {
+    this.once('connect', this._start.bind(this));
+    return;
+  }
+
+  // Socket was destroyed before the connection was established
+  if (!this._handle) return;
+
+  // this._handle.start();
+};
+
+TLSSocket.prototype.setServername = function setServername(
+  this: TLSSocket,
+  name: string
+): void {
+  validateString(name, 'name');
+  // Does not apply to Cloudflare Workers.
+};
+
+TLSSocket.prototype.setSession = function (_session: string | Buffer): void {
+  // Do nothing. We don't support setting session.
+};
+
+// @ts-expect-error TS2322 Inconsistencies between @types/node
+TLSSocket.prototype.getPeerCertificate = function (
+  _detailed?: boolean
+): ReturnType<TLSSocketType['getPeerCertificate']> {
+  // Returns an object representing a peer certificate.
+  // This function is not supported.
+  throw new Error('getPeerCertificate is not implemented');
```

Comment:
Feels weird having `getCertificate()` return `null` but  `getPeerCertificate()` throws.

Reply:
Especially since `getPeerX509Certificate` returns `undefined` and not `null` or throwing lol

Reply:
getPeerCertificate() doesn't return null. It always returns a value in TLS (ref: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/tls.d.ts#L312) but whereas in getCertificate() can return null. That's why I didn't throw an error on getCertificate().

Yes, getPeerCertificate() returns undefined. I explicitly tried to work around with types/node since most people will use types/node in their code. REf: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/tls.d.ts#L437

Reply:
My point here is that this ends up being very inconsistent. We don't implement any of these but one throws, one returns `undefined`, one returns `null`... if this is intended to make it work like Node.js because Node.js happens to have the same quirk, then please add a comment documenting that. Otherwise, I think it's better to err on the side of consistency here for all of the APIs we do not implement.